src/btree/bt_split.c

```diff
@@ -2118,9 +2115,37 @@ __split_multi(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
      * reference structures.
      */
     WT_RET(__wt_calloc_def(session, new_entries, &ref_new));
-    for (i = 0; i < new_entries; ++i)
+    for (i = 0; i < new_entries; ++i) {
         WT_ERR(
           __wt_multi_to_ref(session, page, &mod->mod_multi[i], &ref_new[i], &parent_incr, closing));
+        /*
+         * If there is lookaside information attached to the page being split then there are records
+         * for this page in the lookaside that need to be accounted for in the lookaside information
+         * attached to the split pages. It is too expensive to figure out the exact lookaside
+         * content for each split page separately, so we:
+         * 1. Copy the original page's lookaside information into a split page when the split page
+         *    itself didn't undergo lookaside eviction.
+         * 2. Aggregate the original page's lookaside information with the split page's
+         *    information when split page did undergo lookaside eviction.
+         */
+        if (ref->page_las != NULL) {
+            if (ref_new[i]->page_las == NULL) {
+                WT_ERR(__wt_calloc_one(session, &ref_new[i]->page_las));
+                *ref_new[i]->page_las = *ref->page_las;
+            } else {
+                ref_new[i]->page_las->max_txn =
+                  WT_MAX(ref_new[i]->page_las->max_txn, ref->page_las->max_txn);
+                ref_new[i]->page_las->max_ondisk_ts =
+                  WT_MAX(ref_new[i]->page_las->max_ondisk_ts, ref->page_las->max_ondisk_ts);
+                ref_new[i]->page_las->min_skipped_ts =
+                  WT_MIN(ref_new[i]->page_las->min_skipped_ts, ref->page_las->min_skipped_ts);
+                ref_new[i]->page_las->has_prepares |= ref->page_las->has_prepares;
+            }
+            /* If we haven't re-instantiated this page, this ref should go into lookaside state. */
+            if (ref_new[i]->state != WT_REF_MEM)
+                WT_REF_SET_STATE(ref_new[i], WT_REF_LOOKASIDE);
+        }
+    }
```

Comment:
@sulabhM, this looks to me like it might conflict with the work already in `__wt_multi_to_ref()` that takes the LAS information from the `WT_MULTI` structure. Can we talk about merging the two into a single chunk of code?